Src/library/net/Clib/syncpoll.c

```diff
@@ -107,15 +107,21 @@ EIF_INTEGER c_select_poll(EIF_INTEGER fd)
 
 EIF_INTEGER c_select_poll_with_timeout(EIF_INTEGER fd, 
 		                               EIF_BOOLEAN read_mode,
-									   EIF_INTEGER timeout)
-	/*x Get read/write status for socket fd within `timeout' seconds */
+									   EIF_NATURAL_64 timeout_nano_sec)
+	/*x Get read/write status for socket fd within `timeout_nano_sec` nanoseconds*/
 {
 	fd_set fdmask;
 	struct timeval tmout;
 	int res;
 
-	tmout.tv_sec = (unsigned long) timeout;
-	tmout.tv_usec = (long) 0;
+	tmout.tv_sec = (long) (timeout_nano_sec / RTU64C(1000000000));
```

Comment:
We need to verify that the `timeout_nano_sec` is not going to overflow. On Windows `long` is 32-bit, whereas it is 64-bit on Linux.

Reply:
On the Eiffel side, I added a precondition to always stick to max 32bits.
So this should be safe.
However, I will also modify the C code to ensure no overflow occurs.

Reply:
Just put an assertion

Reply:
maybe safer to have also on C side just in case someone is reusing this C code in the future directly.